Examples/Framework/include/ActsExamples/Utilities/tbbWrap.hpp

```diff
@@ -0,0 +1,169 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+// uncomment to remove all use of tbb library.
+// #define NO_TBB
+
+// #include <iostream>
+
+#ifdef NO_TBB
+#include <stdexcept>
+#else
+#include <tbb/parallel_for.h>
+#include <tbb/queuing_mutex.h>
+#include <tbb/task_arena.h>
+#endif
+
+namespace ActsExamples {
+
+/// Wrapper for most of the tbb functions that we use in Sequencer.
+/// It disables the use of tbb if nthreads=1.
+/// Note that only a small subset of tbb functions are implemented, and
+/// tbb::blocked_range (which doesn't require any thread setup) is still taken
+/// from the tbb library.
+/// However, if NO_TBB is defined, then don't use tbb library at all
+/// (requires nthreads=1 or -1). This allows the ACTS Examples to be built
+/// without the tbb library (and reduces the dependency on ROOT).
+/// Based on an idea from
+///   https://stackoverflow.com/questions/59736661/how-to-completely-switch-off-threading-in-tbb-code
+
+#ifdef NO_TBB
```

Comment:
A way to set this from CMake would be desirable.

Reply:
```suggestion
#ifdef ACTS_NO_TBB
```

Reply:
good idea

Reply:
> A way to set this from CMake would be desirable.

I don't know how the library detection works, so I hope you guys can advise. I was thinking to have an `ACTS_USE_TBB` setting, which would default to `ON` if `ACTS_BUILD_EXAMPLES` is `ON` and the `tbb` library is found.

If `ACTS_USE_TBB=ON`, then it would link to the library. If it is `OFF`, then it would add `-DACTS_NO_TBB` to the compiler options.

Would this work? Is it overkill?

Reply:
Thinking about it, the option should in any case have `EXAMPLES` in the name, to make clear TBB is not used by the core library.

Reply:
CMake option `-DACTS_USE_EXAMPLES_TBB=off` will add `-DACTS_EXAMPLES_NO_TBB` to the compiler options and disable (direct) linking with the `tbb` library. I moved the CMake settings to `Examples/Framework`, since it's only directly used there. 

However I find that `libtbb.so.2` is still linked in, but I think that's because it's used in my versions of DD4hep and ROOT. I tested this works both with (v19.x) and without DD4hep.